### PR TITLE
Drop SIMPLE test and other legacy code [v2]

### PR DESCRIPTION
**Pull request by clebergnu**
This completely removes the SIMPLE test support for the legacy runner, along with some other legacy runner related code.

---

Changes from v1 (#5341):
* Dropped commit [selftests/functional/test_basic.py: split RunnerExecTestStatus](https://github.com/avocado-framework/avocado/pull/5341/commits/69c04e97ed067527aa35f50c89d2bd262520954d) as the tests get removed in a later commit.
* Included "fail reason" for tests returning unsupported status, matching the legacy behavior

### CODE REVIEW

**Review by beraldoleal — COMMENTED** [review]
Hi @clebergnu , this LGTM. Just two small comments, nothing big.

I'm glad to see this getting in shape for a complete removal. Thanks.

File: selftests/functional/test_basic.py [hunk 1]
```diff
@@ -469,7 +469,7 @@ def test_empty_args_list(self):
 
     def test_empty_test_list(self):
         cmd_line = (f'{AVOCADO} run --disable-sysinfo --job-results-dir '
-                    f'{self.tmpdir.name}--test-runner=runner')
+                    f'{self.tmpdir.name} --test-runner=runner')
```
[the remark this answers is not in the document]
Comment (clebergnu) [reply selftests/functional/test_basic.py]: I've squashed that change into the next commit (and added an explanation into the resulting commit).

File: avocado/core/loader.py [hunk 2]
```diff
@@ -44,6 +44,20 @@ class MissingTest:
     """
```
[the remark this answers is not in the document]
Comment (clebergnu) [reply avocado/core/loader.py:46]: Nice catch, thanks! Since the changes were minor, I sent them as a force push.

**Review by richtja — APPROVED** [review]
LGTM, thanks

**Review by beraldoleal — APPROVED** [review]

**Pull request merged**